File: web/src/application/use-cases/recover-active-session.ts
/**
 * RecoverActiveSessionUseCase — GET /api/sessions/active.
 *
 * FR-22 — on page mount we call this; if the backend has an unfinished
 * session for this user we restore the UI to that session (transcripts,
 * speakers, hints rendered visually). The audio capture itself is NOT
 * resumed — the browser requires a user gesture for getDisplayMedia.
 */

import type { Session } from "@/domain/entities/session";
import type { SessionsApiPort } from "@/application/ports/sessions-api.port";

export class RecoverActiveSessionUseCase {
  constructor(private readonly api: SessionsApiPort) {}

  execute(): Promise<Session | null> {
    return this.api.active();
  }
}

File: python_backend/app/application/use_cases/recover_active_session.py
"""FR-22: page-refresh recovery — return latest non-ended session for the user."""
from typing import Optional

from app.application.ports.sessions_repository import SessionsRepository
from app.domain.entities.session import Session


class RecoverActiveSessionUseCase:
    def __init__(self, repo: SessionsRepository) -> None:
        self.repo = repo

    def execute(self, *, user_id: str) -> Optional[Session]:
        return self.repo.get_active_for_user(user_id)
